core/transactions-factories/token_management_transactions_factory.md

```diff
@@ -97,90 +97,164 @@ class TokenManagementTransactionsFactory:
         addRoleESDTTransferRole: bool;
+        addRoleESDTModifyCreator: boolean;
+        addRoleNFTRecreate: boolean;
     }): Transaction;
 
     create_transaction_for_creating_nft({
         sender: IAddress;
-        tokenIdentifier: str;
+        tokenIdentifier: string;
         initialQuantity: int;
-        name: str;
+        name: string;
         royalties: int;
-        hash: str;
+        hash: string;
         attributes: bytes;
-        uris: List[str];
+        uris: List[string];
     }): Transaction;
 
     create_transaction_for_pausing({
         sender: IAddress;
-        tokenIdentifier: str;
+        tokenIdentifier: string;
     }): Transaction;
 
     create_transaction_for_unpausing({
         sender: IAddress;
-        tokenIdentifier: str;
+        tokenIdentifier: string;
     }): Transaction;
 
     create_transaction_for_freezing({
         sender: IAddress;
         user: IAddress;
-        tokenIdentifier: str;
+        tokenIdentifier: string;
     }): Transaction;
 
     create_transaction_for_unfreezing({
         sender: IAddress;
         user: IAddress;
-        tokenIdentifier: str;
+        tokenIdentifier: string;
     }): Transaction;
 
     create_transaction_for_wiping({
         sender: IAddress;
         user: IAddress;
-        tokenIdentifier: str;
+        tokenIdentifier: string;
     }): Transaction;
 
     create_transaction_for_local_minting({
         sender: IAddress;
-        tokenIdentifier: str;
+        tokenIdentifier: string;
         supplyToMint: int;
     }): Transaction;
 
     create_transaction_for_local_burning({
         sender: IAddress;
-        tokenIdentifier: str;
+        tokenIdentifier: string;
         supplyToBurn: int;
     }): Transaction;
 
     create_transaction_for_updating_attributes({
         sender: IAddress;
-        tokenIdentifier: str;
+        tokenIdentifier: string;
         tokenNonce: int;
         attributes: bytes;
     }): Transaction;
 
     create_transaction_for_adding_quantity({
         sender: IAddress;
-        tokenIdentifier: str;
+        tokenIdentifier: string;
         tokenNonce: int;
         quantityToAdd: int;
     }): Transaction;
 
     create_transaction_for_burning_quantity({
         sender: IAddress;
-        tokenIdentifier: str;
+        tokenIdentifier: string;
         tokenNonce: int;
         quantityToBurn: int;
     }): Transaction;
 
-    ...
+    // receiver is the same as the sender
+    create_transaction_for_modifying_royalties({
+        sender: IAddress;
+        tokenIdentifier: string;
+        tokenNonce: int;
+        royalties: int;
+    }): Transaction;
+
+    // receiver is the same as the sender
+    create_transaction_for_setting_new_uris({
+        sender: IAddress;
+        tokenIdentifier: string;
+        tokenNonce: int;
+        uris: List[string]
+    }): Transaction;
+
+    // receiver is the same as the sender
+    create_transaction_for_modifying_creator({
```

Comment:
Just to double check - who is the new creator?

Reply:
based on the PR I added in the description:
```
The creator of a token can be changed. For this, the token has to be moved to the new creator account. The new creator
account requires `ESDTRoleModifyCreator` role. Also, the token has to be of dynamic type.
```